src/Repository/UserRepository.php

```diff
@@ -16,37 +16,33 @@ public function __construct(private DatabaseConnection $connector)
   }
 
 
-  public function createUser(object $user): ?array
+  public function createUser(array $user): ?array
```

Comment:
```suggestion
  public function createUser(string $username, ...): ?User
```

Reply:
Pour les arguments je veux bien mais pour "User" ? Car là le fait de renvoyer "User"  je ne comprend pas vraiment c'est vis à vis de l'objet ou la classe ? Du Model ou Repository ? Si c'est uniquement pour renvoyer l'adresse mail ou le nom d'utilisateur les arguments ne sont pas suffisants ?

 Si l'utilisateur existe déjà il me semble logique de renvoyer le résultat de la requête sql donc un tableau  contenant les éléments qui posent problème puis ensuite de vérifier dans le contrôleur ce qui l'en est.
 
  Sinon étant donné que je me retrouve maintenant avec les arguments autant me servir d'un switch et trois cas  à vérifier : 
- Email & Nom d'utilisateur existent déjà  --> Je renvoi les deux valeurs dans un tableau
- Email existe déjà --> Je renvoi  string email 
- Nom d'utilisateur existe déjà --> Je renvoi la string  nom d'utilisateur

Reply:
```suggestion
 public function createUser(string $username,string $file,string $email, string $password, BackedEnum $userType): ?object
  {

    $dbConnect = $this->connector->connect();
    $test = new UserModel($username,
    $file,
    $email,
    $password,
    $userType);
  default:
      return $test;
    }
```

J'ai cru comprendre qu'il n'était pas possible de retourner la classe elle-même donc je suppose que par retourner User tu parle de l'objet UserModel avec les informations récupérées depuis le model que je retourne ensuite. et dont j'effectue les vérifications dans le contrôleur pour renvoyer les bons message d'erreur ou simplement rediriger vers la page de connexion lorsque tout est bon. 

Est-ce le cas ?